### DISCUSSION TITLE
python extensions not working

### DISCUSSION THREAD

**Discussion by Remearex** [microsoft/pylance-release#5054] [Q&A] [answered]
I have a virtualbox VM, and this vm shares a folder with my actual pc. This folder is what contains all the code. I have vscode on my actual pc to edit that folder's contents, but the code runs on the VM.

The VM is ubuntu 20.04, my actual pc is Windows 10, and the VM's python version is 3.8.10

yesterday, vscode suddenly started crashing. The icon turned into the default windows exe file icon, and whenever I close it, an error window appears. Back then I didn't think much of it, but on second thought, I should have saved that error message. Anyways, I thought that reinstalling vscode would work. After reinstalling, my old extensions are already installed for some reason, but they aren't working. The highlighting and autocomplete functions aren't working, and would display "Loading..." when I hover over stuff:

![autocomplete not working](https://github.com/microsoft/vscode-python/assets/67757989/e6924630-34e8-4bf6-a255-f8efa613c135)

I found a stackoverflow post with a similar issue: https://stackoverflow.com/questions/50389852/visual-studio-code-intellisense-not-working

The post doesn't seem to have a solution, but I tried most of their suggestions. I tried reinstalling the extensions, changing them to pre-release, uninstalling pylance, etc. to no avail. Some replies talked about the path to the python executable, but I feel like that doesn't really apply in my case, since even though I have python on my actual pc, all the code is ran by the python executable on the VM. In any case, I still selected an interpreter at the bottom right of the screen.

Here are my current extensions:

![current extensions](https://github.com/microsoft/vscode-python/assets/67757989/d4df9f32-6191-4040-a95f-bdfa40c03e98)

Here is my current settings.json:

![settings json contents](https://github.com/microsoft/vscode-python/assets/67757989/e61b7940-fb2f-498b-bc14-a744c7376bc3)

A friend suggested looking at the logs for python language server, and there indeed seems to be an error every time I open vscode, though I'm not sure what's causing this:

![python language server log](https://github.com/microsoft/vscode-python/assets/67757989/e3e86f6e-9015-4c96-9ed2-d580b7d50983)

at the end of my ropes here, so any help is greatly appreciated. Please let me know if you need more information.

**Comment by brettcannon** [comment]
Transferred to the Pylance repo.

**Comment by Remearex** [comment, marked as answer]
ok, resolved it (see my reddit post for more info: https://www.reddit.com/r/vscode/comments/17ikibc/python_extensions_not_working/)

but in summary, apparently there are compatibility issues in newer versions of the python and pylance extensions. What I did was to simply uninstall intellicode (though I think it would still work if I didn't uninstall it, I just didn't think I needed the extension), downgrade the python extension to version v2023.16.0, and downgrade the pylance extension to version v2023.9.30

**Reply by debonte** [reply]
Likely a duplicate of #5019.
